udp_bridge/sender.py

```diff
@@ -116,61 +102,81 @@ def validate_params(node:Node):
-        max_queue_size = node.get_parameter('sender_queue_max_size').value
-        topics = node.get_parameter("topics").value
+class UdpBridgeSender:
+    def __init__(self, node: Node):
+        self.node = node
+        self.freq: float = node.get_parameter("send_frequency").value
+
+        params_blackboard = get_parameters_from_other_node(node, "parameter_blackboard", ["monitoring_host_ip"])
+        self.target: str = params_blackboard["monitoring_host_ip"]
+        self.port: int = node.get_parameter("port").value
+        self.sock = self.setup_udp_broadcast_socket()
 
+        topics: list[str] = node.get_parameter("topics").value
+        max_queue_size: int = node.get_parameter("sender_queue_max_size").value
+        message_handler = self.setup_message_handler()
+        self.subscribers: list[AutoSubscriber] = list(
+            map(lambda topic: AutoSubscriber(topic, max_queue_size, message_handler, node), topics)
+        )
+
+    def setup_udp_broadcast_socket(self) -> socket.socket:
         sock = socket.socket(type=socket.SOCK_DGRAM)
         sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-        subscribers = []
+        return sock
+
+    def setup_message_handler(self) -> MessageHandler:
+        encryption_key: Optional[str] = None
+        if self.node.has_parameter("encryption_key"):
+            encryption_key = self.node.get_parameter("encryption_key").value
+
+        return MessageHandler(encryption_key)
 
-        for topic in topics:
-            subscribers.append(AutoSubscriber(topic, max_queue_size, node))
+    def send_messages_in_queue(self):
+        for subscriber in self.subscribers:
+            try:
+                data = subscriber.queue.get_nowait()
 
-        while rclpy.ok:
-            for subscriber in subscribers:
                 try:
-                    data = subscriber.queue.get_nowait()
+                    self.sock.sendto(data + MessageHandler.PACKAGE_DELIMITER, (self.target, self.port))
+                except Exception as e:
+                    self.node.get_logger().error(
+                        f"Could not send data of topic {subscriber.topic} to {self.target} with error {str(e)}"
+                    )
 
-                    for target in targets:
-                        try:
-                            sock.sendto(data + b'\xff\xff\xff', (target, port))
-                        except Exception as e:
-                            node.get_logger().error('Could not send data of topic {} to {} with error {}'
-                                            .format(subscriber.topic, target, str(e)))
+            except Empty:
+                pass
+
+
+def main():
+    rclpy.init()
+    node = Node("udp_bridge_sender", automatically_declare_parameters_from_overrides=True)
+
+    if validate_params(node):
+        sender = UdpBridgeSender(node)
 
-                except Empty:
-                    pass
+        exec = SingleThreadedExecutor()
```

Comment:
Do we REALLY want to use the single threaded executor?!?

Reply:
It seemed to work fine with the `SingleThreadedExecutor`, which also has a lot less overhead compared  to the `MultiThreadedExecutor`.
I'll test some more with the whole stack, but unless it becomes an issue I would prefer to keep the `SingleThreadedExecutor`.